api/src/controllers/anime/index.action.ts

```diff
@@ -0,0 +1,6 @@
+import { Request, Response, NextFunction } from 'express';
+
+async function index(req: Request, res: Response, next: NextFunction) {
```

Comment:
This function is going to need access to pagination params. A middleware will handle parsing that information out of the request from the `routes` definition, but you'll need the pagination types, as well as the pagination utils to handle sending the correct response payload.

```ts
import { 
    Request, 
    NextFunction, 
    PaginatedResponse as PaginatedExpressResponse 
} from '@@types/express.js';
import type { PaginatedResponse } from '@@types/pagination.js';

import { Fortune } from '@@entities/index.js';

import { errors, entities, pagination } from '@@utils/index.js';

async function index(req: Request, res: PaginatedExpressResponse, next: NextFunction) {
    
    const { limit, offset } = res.locals.pagination;

    const [anime, err] = await entities.indexAndCount<Anime>(Anime, {
        limit, offset
    });

    if(err || !anime) {
        return errors.sendEntitiesResponse({
            res,
            err,
            message: "Error finding Anime",
            entityReturn: anime,
            missingEntityReturnMessage: "Unable to find Anime"
        });
    }

    const response: PaginatedResponse<Anime> = pagination.paginateResponse<Anime>(req, res, anime);

    return res.json(response);
};

export default index;
```

Reply:
I dont have Fortune yet in order to import it!